bucket_map/src/bucket.rs

```diff
@@ -402,8 +402,10 @@ impl<'b, T: Clone + Copy + 'static> Bucket<T> {
         data_len: usize,
         ref_count: RefCount,
     ) -> Result<(), BucketMapError> {
+        let num_slots = data_len as u64;
         let best_fit_bucket = MultipleSlots::data_bucket_from_num_slots(data_len as u64);
-        if self.data.get(best_fit_bucket as usize).is_none() {
+        let requires_data_bucket = num_slots > 1 || ref_count != 1;
+        if requires_data_bucket && self.data.get(best_fit_bucket as usize).is_none() {
```

Comment:
Trying to understand the code and the motivation of the PR.

I understand that this will avoid unnecessary access to `self.data.get(best_fit_bucket as usize)`.
But does `self.data.get(best_fit_bucket as usize)` also trigger the bucket file allocation os is it somewhere else?

Reply:
Returning `Err(BucketMapError::DataNoSpace((best_fit_bucket, 0)))` causes us to allocate a data bucket for entry `0` in `data`.
This fn is try, so it fails with an error if reallocation is required. Then the caller resizes, then gets a write lock again, and repeats the try write.

Reply:
resizes means add a data bucket, if the error returned was `DataNoSpace`

Reply:
Thanks for the explanation!